app/upgrades/nextupgrade/upgrades_test.go

```diff
@@ -159,3 +163,56 @@ func (suite *UpgradeTestSuite) TestAdminModuleUpgrade() {
 	suite.Require().NoError(err)
 	suite.Require().Equal(uint64(1), id)
 }
+
+func (suite *UpgradeTestSuite) TestRegisterInterchainAccountCreationFee() {
+	var (
+		app                 = suite.GetNeutronZoneApp(suite.ChainA)
+		ccvConsumerSubspace = app.GetSubspace(ccvconsumertypes.ModuleName)
+		ctx                 = suite.ChainA.GetContext()
+	)
+
+	suite.Require().True(ccvConsumerSubspace.Has(ctx, ccvconsumertypes.KeyRewardDenoms))
+	suite.FundAcc(sdk.AccAddress("neutron1weweewe"), sdk.NewCoins(sdk.NewCoin("untrn", sdk.NewInt(10000))))
+	// emulate mainnet/testnet state
+	ccvConsumerSubspace.Set(ctx, ccvconsumertypes.KeyRewardDenoms, &[]string{params.DefaultDenom})
+
+	var denomsBefore []string
+	ccvConsumerSubspace.Get(ctx, ccvconsumertypes.KeyRewardDenoms, &denomsBefore)
+	suite.Require().Equal(denomsBefore, []string{params.DefaultDenom})
+	contractKeeper := keeper.NewDefaultPermissionKeeper(app.WasmKeeper)
+	// store contract just to increase code_id
+	_ = suite.StoreTestCode(ctx, sdk.AccAddress("neutron1_ica"), "testdata/neutron_interchain_txs.wasm")
+	// store contract for register ica w/o fees
+	codeIDBefore := suite.StoreTestCode(ctx, sdk.AccAddress("neutron1_ica"), "testdata/neutron_interchain_txs.wasm")
+	contractAddressBeforeUpgrade := suite.InstantiateTestContract(ctx, sdk.AccAddress("neutron1_ica"), codeIDBefore)
+
+	upgrade := upgradetypes.Plan{
+		Name:   nextupgrade.UpgradeName,
+		Info:   "some text here",
+		Height: 100,
+	}
+	app.UpgradeKeeper.ApplyUpgrade(ctx, upgrade)
+
+	lastCodeID := app.InterchainTxsKeeper.GetLastCodeIDBeforeUpgrade(ctx)
+	// ensure that wasm module stores next code id
+	suite.Require().Equal(lastCodeID, codeIDBefore+1)
```

Comment:
why you expect the next code if it's named the **last** one?

Reply:
Because the keyword `last` is  inherited from wasm module. Actually, it's more like `next` id